central/credentialexpiry/service/service_impl.go

```diff
@@ -161,6 +170,74 @@ func (s *serviceImpl) getScannerCertExpiry(ctx context.Context) (*v1.GetCertExpi
 	}
 }
 
+func (s *serviceImpl) getScannerV4CertExpiry(ctx context.Context) (*v1.GetCertExpiry_Response, error) {
+	if !features.ScannerV4.Enabled() {
+		return nil, errors.Wrap(errox.InvalidArgs, "Scanner V4 is not integrated")
+	}
+	indexerConfig := s.scannerConfigs[mtls.ScannerV4IndexerSubject]
+	matcherConfig := s.scannerConfigs[mtls.ScannerV4MatcherSubject]
+	if indexerConfig == nil && matcherConfig == nil {
+		return nil, errors.New("could not load TLS configs to talk to Scanner V4 indexer and matcher")
+	}
+
+	integration, exists, err := s.imageIntegrations.GetImageIntegration(ctx, iiStore.DefaultScannerV4Integration.GetId())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve Scanner V4 image integration")
+	}
+	if !exists {
+		return nil, errors.New("Scanner V4 image integration missing")
+	}
+
+	s4Config := integration.GetScannerV4()
+	indexerEndpoint := scannerv4.DefaultIndexerEndpoint
+	if s4Config.GetIndexerEndpoint() != "" {
+		indexerEndpoint = s4Config.GetIndexerEndpoint()
+	}
+	matcherEndpoint := scannerv4.DefaultMatcherEndpoint
+	if s4Config.GetMatcherEndpoint() != "" {
+		matcherEndpoint = s4Config.GetMatcherEndpoint()
+	}
+
+	numEndpoints := 2
+	errC := make(chan error, numEndpoints)
+	expiryC := make(chan *types.Timestamp, numEndpoints)
+	getExpiry := func(subject mtls.Subject, endpoint string) {
+		expiry, err := s.getExpiryFunc(ctx, subject, s.scannerConfigs[subject], endpoint)
+		if err != nil {
+			errC <- err
+			return
+		}
+
+		log.Debugf("Obtained expiry for %q: %s", subject.Identifier, expiry)
+		expiryC <- expiry
+	}
+
+	go getExpiry(mtls.ScannerV4IndexerSubject, indexerEndpoint)
+	go getExpiry(mtls.ScannerV4MatcherSubject, matcherEndpoint)
+
+	errorList := errorhelpers.NewErrorList("failed to determine Scanner V4 cert expiry")
+	expiries := make([]*types.Timestamp, 0, numEndpoints)
+	for i := 0; i < numEndpoints; i++ {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		case err := <-errC:
+			errorList.AddError(err)
+		case expiry := <-expiryC:
+			expiries = append(expiries, expiry)
+		}
+	}
+	if len(expiries) == 0 {
+		return nil, errors.New(errorList.String())
```

Comment:
`errorList.ToError()`

Reply:
Updated. Was keeping this consistent with the rest of the file initially, updated the other usage as well.